fix: fixed release process to use maven publish plugin directly

desusai7 opened the pull request:
### Changes

* Removed usage of oss-gradle-plugin for publishing the package
* updated gradle files to use `maven-publish` plugin for publishing
* updated gradle to version `7.3.3`
* updated release ci actions to use maven publish instead of oss-gradle-plugin
* updated authentication process with maven central to use token instead of passwords

### Testing

Please describe how this can be tested by reviewers. Be specific about anything not tested and reasons why. Since this library has unit testing, tests should be added for new functionality and existing tests should complete without errors.

- [ ] This change adds unit test coverage
- [ ] This change adds integration test coverage
- [x] This change has been tested on the latest version of the platform/language or why not

### Checklist

- [x] I have read the [Auth0 general contribution guidelines](https://github.com/auth0/open-source-template/blob/master/GENERAL-CONTRIBUTING.md)
- [x] I have read the [Auth0 Code of Conduct](https://github.com/auth0/open-source-template/blob/master/CODE-OF-CONDUCT.md)
- [x] All existing and new tests complete without errors

poovamraj reviewed (changes requested):
This looks great! nice work @desusai7, I think we just have to make some changes to the release action as we have changed the number of variables we are passing in and the command to do the release has changed now - https://github.com/auth0/Auth0.Android/blob/main/.github/actions/maven-publish/action.yml#L39

Everything looks good to me

desusai7 commented:
> This looks great! nice work @desusai7, I think we just have to make some changes to the release action as we have changed the number of variables we are passing in and the command to do the release has changed now - https://github.com/auth0/Auth0.Android/blob/main/.github/actions/maven-publish/action.yml#L39
> 
> Everything looks good to me

@poovamraj, updated the maven-publish action with new variables. Can you please take a look again ?

desusai7 commented on .github/actions/maven-publish/action.yml:
```diff
@@ -1,20 +1,19 @@
     required: true
   signing-password:
     required: true
-  java-version:
```
I moved it [here](https://github.com/auth0/Auth0.Android/blob/ecccb8cd9bfdcf944832a8dc1702c4039d5b13d6/.github/actions/maven-publish/action.yml#L4)
